corehq/motech/repeaters/repeater_generators.py

```diff
@@ -415,6 +415,20 @@ def get_test_payload(self, domain):
         return self.get_payload(None, _get_test_form(domain))
 
 
+class FormDictPayloadGenerator(BasePayloadGenerator):
+    format_name = 'form_dict'
+    format_label = _('Python dictionary')
+
+    def get_payload(self, repeat_record, form) -> dict:
+        from corehq.apps.api.util import form_to_es_form
+        es_form = form_to_es_form(form, include_attachments=True)
+        return {'form': es_form.form_data}
+
+    @property
+    def content_type(self):
+        return 'application/x-python'
```

Comment:
Can you explain how this would work? Also, looks like this is JSON based on the next commit

Reply:
Yeah, sorry. (And, yes, `es_form.form_data` is a dictionary that looks like `form['data']`, but massaged for Elasticsearch.)

(The `content_type` is a bit misleading. I defined it because the value it inherits is "text/xml" and that's not right. But it's not used, because this payload is not send immediately as-is. Here's why.)

The original idea behind a payload generator is that it formats a form or a case (or whatever) just before it gets sent off to a remote endpoint in a single request. All the other payload generators have `get_payload()` return a string, to be used as the body of that request. But there are four repeaters that don't work that way, and the FHIRRepeater is one of them.

These four repeaters take the generated payload, deserialize it back into a dictionary, and then parse that dictionary to collect the data they need to execute a workflow of multiple requests.

The older three repeaters use `FormJsonPayloadGenerator`, which massages the form data for Elasticsearch, then **serializes it using Tastypie**, only for the repeater to deserialize it immediately, to get it back to form_json. That seems wasteful! So this payload generator cuts out all the unnecessary processing in the middle.